client-runtime/src/test/java/com/microsoft/rest/v2/http/HttpRequestTests.java

```diff
@@ -12,4 +15,40 @@ public void constructor() {
         assertEquals("request http method", request.httpMethod());
         assertEquals("request url", request.url());
     }
+
+    @Test
+    public void testClone() {
+        final HttpHeaders headers = new HttpHeaders();
+        headers.set("my-header", "my-value");
+        headers.set("other-header", "other-value");
+
+        final HttpRequest request = new HttpRequest(
+                "request caller method",
+                "request http method",
+                "request url",
+                headers,
+                new ByteArrayHttpRequestBody(new byte[0], "application/octet-stream"));
+
+        final HttpRequest clonedRequest = request.clone();
+
+        assertNotEquals(request, clonedRequest);
+        assertNotEquals(request.headers(), clonedRequest.headers());
+
+        assertEquals(request.callerMethod(), clonedRequest.callerMethod());
+        assertEquals(request.httpMethod(), clonedRequest.httpMethod());
+        assertEquals(request.url(), clonedRequest.url());
+
+        assertEquals(request.headers().toMap().size(), clonedRequest.headers().toMap().size());
+        for (HttpHeader clonedHeader : clonedRequest.headers()) {
+            for (HttpHeader originalHeader : request.headers()) {
+                if (clonedHeader == originalHeader) {
+                    fail("Failed to clone HttpHeader instance");
+                }
+            }
+
+            assertEquals(clonedHeader.value(), request.headers().value(clonedHeader.name()));
+        }
+
+        assertEquals(request.body(), clonedRequest.body());
```

Comment:
You might want to add some assertions about same-ness (`assertSame()` or `assertNotSame()`) in this test too.

Reply:
Looked up assertSame and it looks like it indicates more clearly what is under test (the identity of the objects, not the value)